### ISSUE TITLE
Error on upgrade to 2.1

### ISSUE THREAD

**Issue by roglans** [spacepy/spacepy#284]
I am trying to upgrade my spacepy installation from 0.1.6 to 0.2.1 because I'm looking to get the additional pycdf.Library methods. Unfortunately, I am running into an error that I can't figure out. I am running this in a virtualenv with python 3.7. and here is some basic system information:

```
Linux-4.15.0-88-generic-x86_64-with-Ubuntu-18.04-bionic
sys.version_info(major=3, minor=7, micro=1, releaselevel='final', serial=0)
numpy=1.18.1
scipy=1.4.1
matplotlib=3.1.3
```
And the error including the full command output:

```
ERROR: Command errored out with exit status -11: /media/shared/Public/python3_7/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-y3vyl9ga/spacepy/setup.py'"'"'; __file__='"'"'/tmp/pip-install-y3vyl9ga/spacepy/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-243qa8d7/install-record.txt --single-version-externally-managed --compile --install-headers /media/shared/Public/python3_7/include/site/python3.7/spacepy Check the logs for full command output.
    cwd: /tmp/pip-install-y3vyl9ga/spacepy/
    Complete output (137 lines):
    running install
    running build
    C compiler: x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC
    
    creating build
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/spacepy
    creating build/temp.linux-x86_64-3.7/spacepy/libspacepy
    compile options: '-c'
    x86_64-linux-gnu-gcc: spacepy/libspacepy/ctrace2d.c
    x86_64-linux-gnu-gcc: spacepy/libspacepy/poppy.cx86_64-linux-gnu-gcc: spacepy/libspacepy/randomkit.c
    x86_64-linux-gnu-gcc: spacepy/libspacepy/cn_solver.c
    
    x86_64-linux-gnu-gcc: spacepy/libspacepy/toolbox.c
    creating build/lib
    creating build/lib/spacepy
    x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.7/spacepy/libspacepy/ctrace2d.o build/temp.linux-x86_64-3.7/spacepy/libspacepy/poppy.o build/temp.linux-x86_64-3.7/spacepy/libspacepy/randomkit.o build/temp.linux-x86_64-3.7/spacepy/libspacepy/cn_solver.o build/temp.linux-x86_64-3.7/spacepy/libspacepy/toolbox.o -lm -o build/lib/spacepy/libspacepy.so
    running config_fc
    unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options
    running config_cc
    unifing config_cc, config, build_clib, build_ext, build commands --compiler options
    running build_py
    copying spacepy/time.py -> build/lib/spacepy
    copying spacepy/datamodel.py -> build/lib/spacepy
    copying spacepy/omni.py -> build/lib/spacepy
    copying spacepy/seapy.py -> build/lib/spacepy
    copying spacepy/coordinates.py -> build/lib/spacepy
    copying spacepy/spacepy_EnKF.py -> build/lib/spacepy
    copying spacepy/data_assimilation.py -> build/lib/spacepy
    copying spacepy/lib.py -> build/lib/spacepy
    copying spacepy/__init__.py -> build/lib/spacepy
    copying spacepy/datamanager.py -> build/lib/spacepy
    copying spacepy/ae9ap9.py -> build/lib/spacepy
    copying spacepy/radbelt.py -> build/lib/spacepy
    copying spacepy/rst.py -> build/lib/spacepy
    copying spacepy/poppy.py -> build/lib/spacepy
    copying spacepy/empiricals.py -> build/lib/spacepy
    copying spacepy/LANLstar.py -> build/lib/spacepy
    creating build/lib/spacepy/irbempy
    copying spacepy/irbempy/irbempy.py -> build/lib/spacepy/irbempy
    copying spacepy/irbempy/__init__.py -> build/lib/spacepy/irbempy
    creating build/lib/spacepy/pycdf
    copying spacepy/pycdf/istp.py -> build/lib/spacepy/pycdf
    copying spacepy/pycdf/const.py -> build/lib/spacepy/pycdf
    copying spacepy/pycdf/__init__.py -> build/lib/spacepy/pycdf
    creating build/lib/spacepy/plot
    copying spacepy/plot/carrington.py -> build/lib/spacepy/plot
    copying spacepy/plot/apionly.py -> build/lib/spacepy/plot
    copying spacepy/plot/utils.py -> build/lib/spacepy/plot
    copying spacepy/plot/spectrogram.py -> build/lib/spacepy/plot
    copying spacepy/plot/__init__.py -> build/lib/spacepy/plot
    copying spacepy/plot/colourmaps.py -> build/lib/spacepy/plot
    creating build/lib/spacepy/pybats
    copying spacepy/pybats/bats.py -> build/lib/spacepy/pybats
    copying spacepy/pybats/rim.py -> build/lib/spacepy/pybats
    copying spacepy/pybats/trace2d.py -> build/lib/spacepy/pybats
    copying spacepy/pybats/dgcpm.py -> build/lib/spacepy/pybats
    copying spacepy/pybats/interact.py -> build/lib/spacepy/pybats
    copying spacepy/pybats/__init__.py -> build/lib/spacepy/pybats
    copying spacepy/pybats/kyoto.py -> build/lib/spacepy/pybats
    copying spacepy/pybats/batsmath.py -> build/lib/spacepy/pybats
    copying spacepy/pybats/qotree.py -> build/lib/spacepy/pybats
    copying spacepy/pybats/gitm.py -> build/lib/spacepy/pybats
    copying spacepy/pybats/dipole.py -> build/lib/spacepy/pybats
    copying spacepy/pybats/ram.py -> build/lib/spacepy/pybats
    copying spacepy/pybats/pwom.py -> build/lib/spacepy/pybats
    creating build/lib/spacepy/toolbox
    copying spacepy/toolbox/__init__.py -> build/lib/spacepy/toolbox
    creating build/lib/spacepy/data
    copying spacepy/data/spacepy.mplstyle -> build/lib/spacepy/data
    copying spacepy/data/spacepy_altgrid.mplstyle -> build/lib/spacepy/data
    copying spacepy/data/tai-utc.dat -> build/lib/spacepy/data
    copying spacepy/data/spacepy_polar.mplstyle -> build/lib/spacepy/data
    copying spacepy/data/SEA_epochs_OMNI.txt -> build/lib/spacepy/data
    creating build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/Lmax_T89.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/LANLstar_T96.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/LANLstar_T01QUIET.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/LANLstar_T01STORM.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/Lmax_T01STORM.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/Lmax_T96.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/LANLstar_OPDyn.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/LANLstar_T05.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/Lmax_OPDyn.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/Lmax_T01QUIET.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/LANLstar_RAMSCB.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/Lmax_OPQuiet.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/Lmax_T05.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/LANLstar_T89.net.gz -> build/lib/spacepy/data/LANLstar
    copying spacepy/data/LANLstar/LANLstar_OPQuiet.net.gz -> build/lib/spacepy/data/LANLstar
    creating build/lib/spacepy/data/TS07D
    creating build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_13.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_42.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamebhr1.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_43.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_42.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_54.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_11.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_51.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_24.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamebhr4.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_31.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_53.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_23.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_51.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_34.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_52.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_13.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_21.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_31.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_33.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_54.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_14.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamebhr3.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_14.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_32.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_22.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_24.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamebhr5.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_41.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_41.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_22.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_44.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_43.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_12.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_11.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_33.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_44.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_34.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_21.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_12.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_53.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_52.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamebhr2.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_o_32.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    copying spacepy/data/TS07D/TAIL_PAR/tailamhr_e_23.par -> build/lib/spacepy/data/TS07D/TAIL_PAR
    ----------------------------------------```

Any Ideas about what I could do? Many thanks in advance!

**Comment by roglans** [comment]
I have been able to install the dev version of spacepy just by downloading the git master branch, but I was presented with two warnings which have not affected the things that I am doing with spacepy. I'm not sure that they are relevant but here they are anyway:

```
setup produced the following warnings. Some functionality may be missing.

irbemlib module failed. Try a different Fortran compiler? (--fcompiler)
irbemlib build produced no recognizable module. Try a different Fortran compiler? (--fcompiler)
Numpydoc and sphinx required to build documentation.
Help will not be available.

```

**Comment by jtniehof** [comment]
On installing 0.2.1, it would be good to see the full command that's failing (presumably a `pip install`.) There's also nothing particularly useful in that full output; the first line says "Check the logs for full command output" so there might be another log file that's useful.

If you prefer, you should be able to build from the 0.2.1 release files as easily as master (or more easily); those ship with pre-build docs so you wouldn't get the warning about sphinx + numpydoc: https://github.com/spacepy/spacepy/releases

Presuming you're primarily using pycdf, you can ignore the irbem warnings. (It does underly the SpacePy coordinate transforms, in case you need those.) That indicates there isn't a working Fortran compiler installed, which may also be the source of the problems with pip.

In short, I think you have a working install of SpacePy for the purposes you need, but happy to help track down further.

**Comment by jtniehof** [comment]
Any further help we can provide, or shall I close this issue?

**Comment by jtniehof** [comment]
No activity so closing.
